StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1206DeclarationKeywordsMustFollowOrder.cs

```diff
@@ -54,7 +79,177 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+                }
+
+                if (!otherModifiersAppearEarlier)
+                {
+                    otherModifiersAppearEarlier = currentModifierType == ModifierType.Other;
+                }
+
+                previousModifierType = currentModifierType;
+            }
+        }
+
+        private static ModifierType GetModifierType(SyntaxToken modifier)
+        {
+            var result = default(ModifierType);
+
+            switch (modifier.Kind())
+            {
+                case SyntaxKind.PublicKeyword:
+                case SyntaxKind.ProtectedKeyword:
+                case SyntaxKind.InternalKeyword:
+                case SyntaxKind.PrivateKeyword:
+                    result = ModifierType.Access;
+                    break;
+                case SyntaxKind.StaticKeyword:
+                    result = ModifierType.Static;
+                    break;
+                case SyntaxKind.VirtualKeyword:
+                case SyntaxKind.AbstractKeyword:
+                case SyntaxKind.OverrideKeyword:
+                case SyntaxKind.ExternKeyword:
+                case SyntaxKind.UnsafeKeyword:
+                case SyntaxKind.NewKeyword:
+                case SyntaxKind.SealedKeyword:
+                case SyntaxKind.ReadOnlyKeyword:
+                case SyntaxKind.VolatileKeyword:
+                case SyntaxKind.FixedKeyword:
+                case SyntaxKind.ConstKeyword:
+                case SyntaxKind.AsyncKeyword:
+                case SyntaxKind.PartialKeyword:
+                    result = ModifierType.Other;
+                    break;
+            }
+
+            return result;
+        }
+
+        private static int CompareModifiersType(ModifierType first, ModifierType second)
+        {
+            const int lessThan = -1;
+            const int greaterThan = 1;
+
+            var result = 0;
+
+            if (first == second)
+            {
+                result = 0;
+            }
+            else if (first == ModifierType.None)
+            {
+                result = lessThan;
+            }
+            else if (second == ModifierType.None)
+            {
+                result = greaterThan;
+            }
+            else if (first == ModifierType.Access && (second == ModifierType.Static || second == ModifierType.Other))
+            {
+                result = lessThan;
+            }
+            else if (first == ModifierType.Static && second == ModifierType.Other)
+            {
+                result = lessThan;
+            }
+            else if (first == ModifierType.Static && second == ModifierType.Access)
+            {
+                result = greaterThan;
+            }
+            else if (first == ModifierType.Other && (second == ModifierType.Static || second == ModifierType.Access))
+            {
+                result = greaterThan;
+            }
+
+            return result;
+        }
+
+        private static string GetModifierTypeText(ModifierType modifierType)
```

Comment:
:bulb: In practice, I found this had two problems:

1. It makes the messages not localizable, because other languages don't have the same syntax.
2. It makes it more challenging for the reader to know what to move. If it said `internal` must appear before `static`, there would be no question.

I'll file a bug to correct this.